### PR TITLE
[dagster-airlift] test refactors and fixes

### PR DESCRIPTION
**Pull request by dpeng817**
Refactors tests to use the new framework.

### CODE REVIEW

**Comment by dpeng817** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24146?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#24146** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24146?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#24145** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24145?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @dpeng817 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: examples/experimental/dagster-airlift/dagster_airlift/test/airflow_test_instance.py [hunk 1]
```diff
@@ -127,5 +136,44 @@ def make_dag_run(dag_id: str, run_id: str, start_date: datetime, end_date: datet
...
+            "conf": {},
         },
     )
+
+
+def make_instance(
+    dag_and_task_structure: Dict[str, List[str]],
+    dag_runs: List[DagRun] = [],
+) -> AirflowTestInstance:
+    """Return a single dag instance with a set of tasks. No runs or task instances are included."""
```
Comment (schrockn) [inline examples/experimental/dagster-airlift/dagster_airlift/test/airflow_test_instance.py:150]: This docblock is a little misleading. It creates task instances and it takes passed in runs

File: examples/experimental/dagster-airlift/dagster_airlift_tests/unit_tests/conftest.py [hunk 2]
```diff
@@ -1,210 +1,72 @@
...
-
-        class ListDagsOverride(klass_to_instantiate):  # type: ignore
-            def list_dags(self):
-                return list_dags_override(self)
-
-        klass_to_instantiate = ListDagsOverride
-
-    return klass_to_instantiate()
-
-
```
Comment (schrockn) [inline examples/experimental/dagster-airlift/dagster_airlift_tests/unit_tests/conftest.py:165]: nice to see this go

**Review by schrockn — APPROVED** [review]
Excellent. It is wise to invest a bunch in this test suite. There will be a very long tail of scenarios we want to test and it is untenable to do them all in integration tests given their runtime.

File: examples/experimental/dagster-airlift/dagster_airlift/test/airflow_test_instance.py (hunk 1 above)
Comment (dpeng817) [reply examples/experimental/dagster-airlift/dagster_airlift/test/airflow_test_instance.py:150]: Ah whoops. Docblock from a previous iteration.

**Pull request merged**